core/src/main/scala/caliban/Rendering.scala

```diff
@@ -79,6 +79,44 @@ object Rendering {
       }
       .mkString("\n\n")
 
+  def renderDirectives(directives: List[__Directive]): String =
+    directives.map(renderDirective).mkString("\n")
+
+  private def renderDirective(directive: __Directive): String = {
+    val inputs             = directive.args match {
+      case i if i.nonEmpty => s"""(${i.map(renderInputValue).mkString(", ")})"""
+      case _               => ""
+    }
+    val locationStrings    = directive.locations.map {
+      case __DirectiveLocation.QUERY                  => "QUERY"
+      case __DirectiveLocation.MUTATION               => "MUTATION"
+      case __DirectiveLocation.SUBSCRIPTION           => "SUBSCRIPTION"
+      case __DirectiveLocation.FIELD                  => "FIELD"
+      case __DirectiveLocation.FRAGMENT_DEFINITION    => "FRAGMENT_DEFINITION"
+      case __DirectiveLocation.FRAGMENT_SPREAD        => "FRAGMENT_SPREAD"
+      case __DirectiveLocation.INLINE_FRAGMENT        => "INLINE_FRAGMENT"
+      case __DirectiveLocation.SCHEMA                 => "SCHEMA"
+      case __DirectiveLocation.SCALAR                 => "SCALAR"
+      case __DirectiveLocation.OBJECT                 => "OBJECT"
+      case __DirectiveLocation.FIELD_DEFINITION       => "FIELD_DEFINITION"
+      case __DirectiveLocation.ARGUMENT_DEFINITION    => "ARGUMENT_DEFINITION"
+      case __DirectiveLocation.INTERFACE              => "INTERFACE"
+      case __DirectiveLocation.UNION                  => "UNION"
+      case __DirectiveLocation.ENUM                   => "ENUM"
+      case __DirectiveLocation.ENUM_VALUE             => "ENUM_VALUE"
+      case __DirectiveLocation.INPUT_OBJECT           => "INPUT_OBJECT"
+      case __DirectiveLocation.INPUT_FIELD_DEFINITION => "INPUT_FIELD_DEFINITION"
+    }
+    val directiveLocations = locationStrings.mkString(" | ")
+
+    val body = s"""directive @${directive.name}${inputs} repeatable on ${directiveLocations}""".stripMargin
```

Comment:
Should we add `repeatable: Boolean` to directives with default to true?

Reply:
Oh shoot I meant to ask about this. Right now it's hard-coded as `repeatable`. I didn't look at what we did for this with the introspection API but it would be good to allow it to be set as non-repeatable.

Reply:
Ah this keyword is new in the 2021 spec, which is why it was not there before. Would be nice to add (otherwise, don't include the keyword).

Reply:
Cool, let me see what I can do about adding it.

Reply:
On the top of my head, it should be added in the parser, in the `Directive` and `__Directive` classes, and maybe a validation rule should be added to check there is only one given if not repeatable.

If that's too much, it can be done in another PR 😄

Reply:
Cool. Yeah I will remove the `repeatable` statement for now, and then do another PR to add `repeatable`.